corehq/apps/app_manager/suite_xml.py

```diff
@@ -1504,14 +1504,16 @@ def get_new_case_id_datums_meta(self, form):
                 })
 
             if 'subcases' in actions:
-                non_repeat_actions = [a for a in actions['subcases'] if not a.repeat_context]
-                for i, subcase in enumerate(non_repeat_actions):
-                    datums.append({
-                        'datum': SessionDatum(id=form.session_var_for_action('subcase', i), function='uuid()'),
-                        'case_type': subcase.case_type,
-                        'requires_selection': False,
-                        'action': subcase
-                    })
+                for i, subcase in enumerate(actions['subcases']):
+                    # don't put this in the loop to be consistent with the form's indexing
+                    # see XForm.create_casexml_2
+                    if not subcase.repeat_context:
+                        datums.append({
+                            'datum': SessionDatum(id=form.session_var_for_action('subcase', i), function='uuid()'),
+                            'case_type': subcase.case_type,
+                            'requires_selection': False,
+                            'action': subcase
+                        })
```

Comment:
this is the only change. basically when the form generates it session IDs it does so indexing over the entire list of subcase actions, but when the suite does it it was pruning them off first. as a result if you had repeats mixed with regular cases the indices would be off-by-one (or however many repeats you had)